ui/app/components/auth/form-template.hbs

```diff
@@ -0,0 +1,105 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+~}}
+
+{{#if this.formComponent}}
+  {{#let (component this.formComponent) as |AuthFormComponent|}}
```

Comment:
Open to alternative solutions here, this seemed like a decent way to separate the display logic (handling hiding/showing tabs) from the login/auth logic which will live in either `auth/form/base.js` or the corresponding auth method component file for more complex login flows.

See `okta.js` [in the POC PR as an example](https://github.com/hashicorp/vault/pull/29892/files#diff-a565926d76e26c14f8de059f4a67c1a850153a6d03f284425b96729eb9555002)

Reply:
I think this works! It's a clean way to separate those pieces out!